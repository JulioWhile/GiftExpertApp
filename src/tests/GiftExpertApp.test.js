import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GiftExpertApp } from '../GiftExpertApp';

describe('Pruebas en <GiftExpertApp />', () => {
	test('debe coincidir con el snapshot', () => {
		const wrapper = shallow(<GiftExpertApp />);
		expect(wrapper).toMatchSnapshot();
	});

	test('debe de mostrar una lista de categorias', () => {
		const categories = ['One Punch', 'Dragon Ball'];
		const wrapper = shallow(
			<GiftExpertApp defaultCategories={categories} />
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GifGrid').length).toBe(categories.length);
	});
});
