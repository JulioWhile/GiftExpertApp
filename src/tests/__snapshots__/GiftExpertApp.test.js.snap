// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftExpertApp /> debe coincidir con el snapshot 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GiftExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
